client/src/components/posts/PostList.jsx

```diff
@@ -15,6 +18,13 @@ export const PostList = ({ loggedInUser }) => {
         getUnapprovedCount().then(setUnapprovedCount)
     }, [])
 
+    useEffect(() => {
+        getAllCategories().then(setCategories)
```

Comment:
Move the logic in this useEffect to the useEffect that is also getting all posts. There is no reason they need to be in separate useEffects